tests/components/purpleair/test_config_flow.py

```diff
@@ -143,3 +144,129 @@ async def test_reauth(
     assert result["type"] == data_entry_flow.FlowResultType.ABORT
     assert result["reason"] == "reauth_successful"
     assert len(hass.config_entries.async_entries()) == 1
+
+
+@pytest.mark.parametrize(
+    "get_nearby_sensors_mock,get_nearby_sensors_errors",
+    [
+        (AsyncMock(return_value=[]), {"base": "no_sensors_near_coordinates"}),
+        (AsyncMock(side_effect=Exception), {"base": "unknown"}),
+        (AsyncMock(side_effect=PurpleAirError), {"base": "unknown"}),
+    ],
+)
+async def test_options_add_sensor(
+    hass,
+    api,
+    config_entry,
+    get_nearby_sensors_errors,
+    get_nearby_sensors_mock,
+    setup_purpleair,
```

Comment:
I noticed the `setup_purpleair` fixture uses `async_setup_component` and passes a config. That's not useful. This integration only uses config entries, so it's better to use `hass.config_entries.async_setup` to set up the config entry.

Reply:
But I don't always want the config entry to be set up automatically. Using `hass.config_entries.async_setup` will cause most of my tests to fail because they'll be attempting to duplicate an already-existing config entry. `async_setup_component` ensures that I can detach platform setup (including entities, devices, etc.) from where the config entry exists.

Reply:
I'm not following. `hass.config_entries.async_setup` will set up a config entry, it doesn't create a config entry.

Reply:
Where do I get that config entry ID from? From the existing `config_entry` fixture?

Reply:
Yes. 👍

Reply:
Ah, okay—then, in my config flow tests, I can just make sure to use different unique data to avoid duplicates. Thank you!

Reply:
https://github.com/home-assistant/core/pull/84894